config/printer-creality-ender3-s1-2021.cfg

```diff
@@ -0,0 +1,132 @@
+# This file contains pin mappings for the stock 2021 Creality Ender 3
+# S1. To use this config, check the STM32 Chip on the V2.4S1 Board
+# then during "make menuconfig" select either the STM32F103 with a
+# "28KiB bootloader" or select the STM32F401 with a "64KiB bootloader"
+# and serial (on USB PA10/PA9) communication for both depending on the
+# STM32 Chip installed on your printers Motherboard.
+
+# If you prefer a direct serial connection, in "make menuconfig"
+# select "Enable extra low-level configuration options" and select
+# serial (on USB PA10/PA9), which is broken out on the 10 pin IDC
+# cable used for the LCD module as follows:
+# 3: Tx, 4: Rx, 9: GND, 10: VCC
+
+# Flash this firmware by copying "out/klipper.bin" to a SD card and
+# turning on the printer with the card inserted. The firmware
+# filename must changed to "firmware.bin"
+
+# See docs/Config_Reference.md for a description of parameters.
+
+[stepper_x]
+step_pin: PC2
+dir_pin: PB9
+enable_pin: !PC3
+microsteps: 16
+rotation_distance: 40
+endstop_pin: !PA5
+position_endstop: 0
```

Comment:
Endstop is outside printbed so printing at X0 prints in the air. Suggest setting at -7 to match stock firmware postition.

Reply:
Did you also check y axis? I had to set y endstop to -5 to prevent Y0 being in the air. Maybe there is also a value in the stock firmware?